libs/application/ui-fields/src/lib/CompanySearchFormField/CompanySearchFormField.tsx

```diff
@@ -0,0 +1,135 @@
+import { gql, useLazyQuery } from '@apollo/client'
+import {
+  CompanySearchField,
+  FieldBaseProps,
+  formatText,
+} from '@island.is/application/core'
+import { Box } from '@island.is/island-ui/core'
+import { useLocale } from '@island.is/localization'
+import debounce from 'lodash/debounce'
+import React, { FC, useMemo, useState } from 'react'
+import { CompanySearchItem } from './CompanySearchItem'
+import kennitala from 'kennitala'
+import { CompanySearchController } from '@island.is/shared/form-fields'
+import { mockCompanies } from './__mock-data__/companies'
+
+export const COMPANY_REGISTRY_COMPANIES = gql`
+  query SearchCompanies($input: RskCompanyInfoSearchInput!) {
+    companyRegistryCompanies(input: $input) {
+      data {
+        name
+        nationalId
+      }
+    }
+  }
+`
+
+interface Props extends FieldBaseProps {
+  field: CompanySearchField
+}
+
+export const CompanySearchFormField: FC<Props> = ({ application, field }) => {
+  const {
+    id,
+    title,
+    placeholder,
+    setLabelToDataSchema = true,
+    useMockOptions = false,
+  } = field
+  const { formatMessage } = useLocale()
+
+  const [searchQuery, setSearchQuery] = useState('')
+  const [mockData, setMockData] = useState(mockCompanies)
+
+  const [search, { loading, data }] = useLazyQuery(COMPANY_REGISTRY_COMPANIES)
+
+  const defaultAnswer = { value: '', label: '' }
+  const initialValue = (application.answers[id] || { ...defaultAnswer }) as {
+    value: string
+    label: string
+  }
+
+  const debouncer = useMemo(() => {
+    return debounce(search, 250)
```

Comment:
250ms feels a bit low so looked and found that other search inputs in the monorepo have 300ms.

Would be nice to have this in a constant variable. Even better for it to be shared.

Reply:
Made 300ms a shared constant